### ISSUE TITLE
015 Anomaly Detection Ml Model

### ISSUE THREAD

**Issue by simplysandeepp** [simplysandeepp/claimHeart#27]
## 🎯 Issue Description

Implement Isolation Forest ML model for detecting anomalous claim patterns that don't match clear rules but are statistically suspicious.

## 📋 Current Behavior

- Only rule-based fraud detection
- Cannot detect subtle anomalies
- No ML-based scoring

## ✅ Expected Behavior

- Isolation Forest model trained on claim data
- Anomaly score (0-100) for each claim
- Detects unusual patterns
- Works alongside rule engine

## 🔧 Technical Details

**Files to Create:**
- `backend/app/ml/isolation_forest.py`
- `backend/app/ml/model_registry.py`

**Features for Model:**
- Cost amount
- Number of tests
- Hospital stay duration
- Claim frequency
- Time since policy start
- Age of patient

**Implementation Checklist:**
- [ ] Prepare training data
- [ ] Train Isolation Forest model
- [ ] Save model to disk
- [ ] Create prediction function
- [ ] Return anomaly score
- [ ] Integrate with fraud agent

## 💡 Suggested Implementation

```python
from sklearn.ensemble import IsolationForest
import joblib

class AnomalyDetector:
    def __init__(self):
        self.model = joblib.load("models/isolation_forest.pkl")
    
    def get_anomaly_score(self, claim_features: dict) -> float:
        features = [
            claim_features["cost"],
            claim_features["test_count"],
            claim_features["duration_days"],
            claim_features["claim_frequency"],
            claim_features["days_since_policy_start"]
        ]
        
        score = self.model.decision_function([features])[0]
        # Convert to 0-100 scale
        anomaly_score = max(0, min(100, (1 - score) * 50))
        return anomaly_score
```

## ✅ Acceptance Criteria

- [ ] Model trained successfully
- [ ] Returns anomaly score 0-100
- [ ] Detects unusual patterns
- [ ] Low false positive rate
- [ ] Integrated with fraud agent
- [ ] Model can be retrained

## 🏷️ Labels

- `backend`
- `ml`
- `fraud-agent`
- `phase-4`

**Priority:** Medium  
**Estimated Effort:** 6-8 hours  
**Phase:** Phase 4
